src/main/java/seedu/address/logic/commands/FacebookPostCommand.java

```diff
@@ -30,22 +35,34 @@ public FacebookPostCommand(String message) {
         toPost = message;
     }
 
-    @Override
-    public CommandResult execute() throws CommandException {
-        if (!FacebookConnectCommand.isAuthenticated()) {
-            FacebookConnectCommand newFacebookConnect = new FacebookConnectCommand();
-            newFacebookConnect.execute();
-        }
-
+    /**
+     * Completes the post command
+     * @throws CommandException
+     */
+    public static void completePost() throws CommandException {
         Facebook facebookInstance = FacebookConnectCommand.getFacebookInstance();
-        String user = "";
+        user = "";
         try {
             user = facebookInstance.getName();
             facebookInstance.postStatusMessage(toPost);
         } catch (FacebookException e) {
             e.printStackTrace();
         }
```

Comment:
Can we handle the error here? Maybe throw an exception? Since we should not continue on if it fails.

Reply:
Ah ok this is abit tricky cause the post goes through but sometimes it still throws an exception, so im planning to leave it first